test/PushNotificationTests/Shared.cpp

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include <pch.h>
+#include <TestDef.h>
+#include "Shared.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt;
+using namespace winrt::Windows::ApplicationModel;
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+
+namespace Test::PushNotifications
+{
+    StorageFile CreateDocFile(std::wstring filename)
+    {
+        return KnownFolders::DocumentsLibrary().CreateFileAsync(filename,
+            CreationCollisionOption::OpenIfExists).get();
+    }
+
+    StorageFile OpenDocFile(std::wstring filename)
+    {
+        return KnownFolders::DocumentsLibrary().GetFileAsync(filename).get();
+    }
+
+    wil::unique_handle Execute(const std::wstring& command, const std::wstring& args,
+        const std::wstring& directory)
+    {
+        SHELLEXECUTEINFO ei{};
+        ei.cbSize = sizeof(SHELLEXECUTEINFO);
+        ei.fMask = SEE_MASK_NOCLOSEPROCESS | SEE_MASK_DOENVSUBST;
+        ei.lpFile = command.c_str();
+        ei.lpParameters = args.c_str();
+        ei.lpDirectory = directory.c_str();
+
+        if (!ShellExecuteEx(&ei))
+        {
+            auto lastError = GetLastError();
+            VERIFY_WIN32_SUCCEEDED(lastError);
+        }
+
+        wil::unique_handle process{ ei.hProcess };
+        return process;
+    }
+
+    void RunCertUtil(const std::wstring& path, bool removeCert)
```

Comment:
Can you check with @DrusTheAxe to see if this cert/package stuff should be elevated to a more common location?

Reply:
@DrusTheAxe Any ideas where this can be moved?

Reply:
Don't do this.

Adding certs requires admin privilege. RunAs:RestrictedUser can't do it, and making CLASS_SETUP() RunAs:ElevatedUser often mixes "class setup stuff needing admin and class setup stuff that doesn't" with negative downstream fallout (e.g. tests cannot run /inproc).

Standard guidance to date is "The test pipeline runs tools\DevCheck.cmd to add the test cert if necessary, among other things. Tests don't do this; it's part of their environmental soup they expect to run in".

Developers run tools\DevCheck.cmd to develop locally (typically run once, then only needed if things change -- if a cert expires and is renewed, if TAEF's version changes, etc).

Test environments (Helix etc) do the same **using the same tool**.

This consistency also ensures developers are working with the same infrastructure the pipeline's tests will also expect/provide. (We spent waaaaay too much time with inner-loop + pipeline-test using totally disjointed logic).

TL;DR delete `RunCertUtil()` and `Execute()` and rely on tools\DevCheck.cmd